drivers/adodb-mysqli.inc.php

```diff
@@ -231,9 +231,13 @@ function _connect($argHostname = null,
 
 		// SSL Connections for MySQLI
 		if ($this->ssl_key || $this->ssl_cert || $this->ssl_ca || $this->ssl_capath || $this->ssl_cipher) {
+
 			mysqli_ssl_set($this->_connectionID, $this->ssl_key, $this->ssl_cert, $this->ssl_ca, $this->ssl_capath, $this->ssl_cipher);
-			$this->socket = MYSQLI_CLIENT_SSL;
-			$this->clientFlags = MYSQLI_CLIENT_SSL_DONT_VERIFY_SERVER_CERT;
+
+			// Default client flag to ssl with verify if not set.
+			if (0 === $this->clientFlags) {
+				$this->clientFlags = MYSQLI_CLIENT_SSL_VERIFY_SERVER_CERT;
```

Comment:
Considering that clientFlags is a binary flag, a straight assignment will potentially overwrite the user's flags.

You should OR the value in `$this->clientFlags |= MYSQLI_CLIENT_SSL_VERIFY_SERVER_CERT;` instead